core/src/test/java/hudson/model/ViewTest.java

```diff
@@ -112,6 +113,62 @@ private TopLevelItem createJob(String jobName) {
         return rootJob;
     }
 
+    @Test
+    public void buildQueueFiltering() throws Exception {
+        // Mimic a freestyle job
+        FreeStyleProject singleItemJob = Mockito.mock(FreeStyleProject.class);
+        Mockito.when(singleItemJob.getOwnerTask()).thenReturn(singleItemJob);
+        Queue.Item singleItemQueueItem = new MockItem(singleItemJob);
+
+        // Mimic pattern of a Matrix job, i.e. with root item in view and sub
+        // items in queue.
+        FreeStyleProject multiItemJob = Mockito.mock(FreeStyleProject.class);
+        Project multiItemSubJob = Mockito.mock(Project.class);
+        Mockito.when(multiItemSubJob.getRootProject()).thenReturn(multiItemJob);
+        Mockito.when(multiItemSubJob.getOwnerTask()).thenReturn(multiItemSubJob);
+        Queue.Item multiItemQueueItem = new MockItem(multiItemSubJob);
+
+        // Mimic pattern of a Pipeline job, i.e. with item in view and
+        // sub-steps in queue.
```

Comment:
Fine although I generally do not trust Mockito/PowerMock tests at all; you anyway need to go back and confirm the actual behavior (in this case using a downstream PR in `workflow-durable-task-step-plugin`).

Reply:
Okay, will do! Is it ok to add an actual Jenkins instance test in the downstream plugin? Sure feels more robust that way!